js/utils.js

```diff
@@ -0,0 +1,45 @@
+async function makeApiCall(
+  url,
+  method = 'get',
+  body = null,
+  credentials = 'include',
+  headers = { 'content-type': 'application/json' },
+  options = null,
+) {
+  try {
+    const response = await fetch(url, {
+      method,
+      body,
+      headers,
+      credentials,
+      ...options,
+    });
+    const data = await response.json();
+    const status = response.status;
+    const res = {
+      data,
+      status,
+    };
+    return res;
+  } catch (err) {
```

Comment:
any reason why you are using `throw ` in catch block? as it will throw the error to its superior scope. are you wanting your superior scope function to do something very specific when this error is thrown?

Reply:
I am using makeApiCall at two place but it is not using throw error very specific...So if it is not using should  I using console.log(err) here??

1. https://github.com/Real-Dev-Squad/website-www/pull/432/files#diff-b8034978785404c0d5db1ed2aa00f41fecf7baaf67f71b348a0a1806a5b24017R255

2.  https://github.com/Real-Dev-Squad/website-www/pull/432/files#diff-b8034978785404c0d5db1ed2aa00f41fecf7baaf67f71b348a0a1806a5b24017R291

Reply:
`console.error` should be a better alternative

Reply:
vaibhav I am using `throw err` because utils.js file is a global file and it depend on particular user how they want to handle error in api call error...For ex: I am handle error by showing toast.

Reply:
but then are you even handling the error that is thrown in the superior scope?

Reply:
yes I am handling that error if you see changes now...but also Have question correct me if i am wrong
Its depend on developer to developer that they want to show that error or not?